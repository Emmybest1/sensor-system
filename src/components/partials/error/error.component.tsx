import React from 'react';
import styled from 'styled-components';

const ErrorNode = styled.div`
  color: #f87272;
  font-size: 0.9rem;
  width: 100%;
`;

type ErrorPropTypes = {
  message: string;
};

export const Error: React.FC<ErrorPropTypes> = ({ message }) => (
  <ErrorNode aria-live="assertive" aria-atomic="true">
    {message}
  </ErrorNode>
);
