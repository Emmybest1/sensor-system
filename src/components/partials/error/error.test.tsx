import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Error } from './error.component';
import { ReduxProvider } from 'redux/__test__/redux-state.mock';

describe('<Error/>', () => {
  test('should render Component', () => {
    const { container } = render(
      <ReduxProvider internal>
        <Router>
          <Error message="Sorry you can't proceed" />
        </Router>
      </ReduxProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
