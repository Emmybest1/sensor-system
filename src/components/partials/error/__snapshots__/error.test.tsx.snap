// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> should render Component 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  class="sc-bdfBQB eXRiR"
>
  Sorry you can't proceed
</div>
`;
